src/logic/basic.lean

```diff
@@ -1599,4 +1605,9 @@ by by_cases h : P; simp [h]
 lemma ite_and : ite (P ∧ Q) a b = ite P (ite Q a b) b :=
 by by_cases hp : P; by_cases hq : Q; simp [hp, hq]
 
+lemma ite_ite_comm_of_ne {α β} [decidable_eq β] {a b c : α} {i j k : β} (h : j ≠ k) :
+  (if i = j then a else if i = k then b else c) =
+  if i = k then b else if i = j then a else c :=
```

Comment:
Does this generalize to something like
```suggestion
lemma ite_ite_comm_of_disjoint {α β} [decidable p] [decidable q] {a b c : α} (h : disjoint p q) :
  (if p then a else if q then b else c) =
  if q then b else if p then a else c :=
```

Reply:
Yes it generalizes. However `disjoint` isn't imported, and probably shouldn't be in such a basic file. `disjoint p q` probably defeq to `¬ (p ∧ q)` (then should the name be `ite_ite_comm_of_not_and`?), or should I use `p → ¬q`?

Reply:
`ite_ite_comm_of_not_and` sounds good to me. Maybe it can even just be called `ite_ite_comm` now?
Concerning `disjoint`: please add `disjoint.ite_ite_comm` in the relevant file.

Reply:
Thanks! I've added `dite_dite_comm` and renamed the existing one to `ite_ite_comm`, using `P → ¬Q` for the spelling of the disjoint condition. I'm not in favor of adding a version using `disjoint`: (1) that spelling is longer; (2) you need to destruct `and` when supplying that condition; (3) the file where `disjoint` and [Prop.distrib_lattice](https://leanprover-community.github.io/mathlib_docs/order/bounded_order.html#Prop.distrib_lattice) are defined is about order (*order/bounded_order*), and feels like a weird place to add lemmas about `ite`.

Reply:
Ok, fair enough.